[2.0.2] *: add `transfer-source` command (#605)
pingcap/docs-dm: en/manage-source.md

}
```

If you don't know the `source-id`, you could use `dmctl --master-addr <master-addr> operate-source show` to list all data sources.

{{< copyable "" >}}

```bash
operate-source show
```

```
{
    "result": true,
    "msg": "",
    "sources": [
        {
            "result": true,
            "msg": "source is added but there is no free worker to bound",
            "source": "mysql-replica-02",
            "worker": ""
        },
        {
            "result": true,
            "msg": "",
            "source": "mysql-replica-01",
            "worker": "dm-worker-1"
        }
    ]
}
```

> ## Change the binding relations of upstream MySQL instances and DM-workers

Reviewer: ```suggestion
## Change the bindings between upstream MySQL instances and DM-workers
```

> You can use the `transfer-source` command change the binding relations of upstream MySQL instances and DM-workers.

Reviewer: ```suggestion
You can use the command `transfer-source` command to change the bindings between upstream MySQL instances and DM-workers.
```

{{< copyable "" >}}

```bash
help transfer-source
```

```

> Transfers a upstream MySQL/MariaDB source to a free worker.

Reviewer: ```suggestion
Transfers an upstream MySQL/MariaDB source to a free worker.
```

Usage:
  dmctl transfer-source <source-id> <worker-id> [flags]
Flags:
  -h, --help   help for transfer-source
Global Flags:
  -s, --source strings   MySQL Source ID.
```

> Before changing, DM will check if the unbinding worker is still has running tasks. If so, please [pause tasks](pause-task.md) before the changing, and [resume tasks](resume-task.md) after the changing.

Reviewer: ```suggestion
Before transferring, DM checks whether the free worker still has running tasks. If the worker does have running tasks, you need to [pause tasks](pause-task.md) before changing the binding and [resume tasks](resume-task.md) after changing the binding.
```

Author: not "the free worker", will change description later

> Before transferring, DM checks whether the worker to be unbound still has running tasks. If the worker does have running tasks, you need to [pause tasks](pause-task.md) before changing the binding and [resume tasks](resume-task.md) after changing the binding.

Author: should we use `[resume the tasks]` with `[pause the tasks]`

> Before transferring, DM checks whether the worker to be unbound still has running tasks. If the worker does have running tasks, you need to [pause tasks](pause-task.md) before changing the binding and [resume tasks](resume-task.md) after changing the binding.

Reviewer 2: Yes, adding "the" is better

### Usage example

> If you don't know the binding relation of DM-workers, you could use `dmctl --master-addr <master-addr> list-member --worker` to list current binding relations.

Reviewer: ```suggestion
If you do not know the bindings of DM-workers, you can run `dmctl --master-addr <master-addr> list-member --worker` to list the current bindings of all workers.
```

{{< copyable "" >}}

```bash
list-member --worker
```

```
{
    "result": true,
    "msg": "",
    "members": [
        {
            "worker": {
                "msg": "",
                "workers": [
                    {
                        "name": "dm-worker-1",
                        "addr": "127.0.0.1:8262",
                        "stage": "bound",
                        "source": "mysql-replica-01"
                    },
                    {
                        "name": "dm-worker-2",
                        "addr": "127.0.0.1:8263",
                        "stage": "free",
                        "source": ""
                    }
                ]
            }
        }
    ]
}
```

> In this example `mysql-replica-01` is bound to `dm-worker-1`. The below command should transfer the source to bind to `dm-worker-2`.

Reviewer: ```suggestion
In the above example, `mysql-replica-01` is bound to `dm-worker-1`. The below command transfers the binding worker of `mysql-replica-01` to `dm-worker-2`.
```

{{< copyable "" >}}

```bash
transfer-source mysql-replica-01 dm-worker-2
```

```
{
    "result": true,
    "msg": ""
}
```

> Check the command has taken effect by `dmctl --master-addr <master-addr> list-member --worker`.

Reviewer: ```suggestion
Check whether the command takens effect by `dmctl --master-addr <master-addr> list-member --worker`.
```
